src/libpmem2/badblocks_ndctl.c

```diff
@@ -739,51 +691,48 @@ badblocks_clear(const char *file, struct badblocks *bbs)
 
 /*
  * badblocks_clear_all -- clears all bad blocks in a file
- *                           (regular file or dax device)
+ *                        (regular file or dax device)
  */
 int
 badblocks_clear_all(const char *file)
 {
 	LOG(3, "file %s", file);
 
-	os_stat_t st;
-	if (os_stat(file, &st) < 0) {
-		ERR("!stat %s", file);
-		return -1;
-	}
-
-	enum pmem2_file_type pmem2_type;
+	struct pmem2_source src;
```

Comment:
Again, this shouldn't exist.

Reply:
What shouldn't exist? ```struct pmem2_source src```?

Reply:
this entire function, it's only ever used externally. 
but I guess this can be done as a next step.

Reply:
Yes, it will be done as a next step in a separate PR.